src/server/list_family.cc

```diff
@@ -265,7 +262,9 @@ OpResult<string> OpMoveSingleShard(const OpArgs& op_args, string_view src, strin
     dest_it->second.ImportRObj(obj);
 
     // Insertion of dest could invalidate src_it. Find it again.
-    src_it = db_slice.GetTables(op_args.db_cntx.db_index).first->Find(src);
+    src_res = db_slice.FindMutable(op_args.db_cntx, src, OBJ_LIST);
```

Comment:
on this copy you will run the original src_res post update which might hold invalid iterator

Reply:
You're very much right!
I added a `DCHECK()` to make sure that the DB is not changed (based on its size).
I think that this is another point _for_ this PR, as these kinds of checks help reduce bugs.

If you can think of other cases we should `DCHECK()` please let me know!